src/lib/HST/Frontend/Syntax.hs

```diff
@@ -78,9 +78,9 @@ deriving instance ShowAST a => Show (Decl a)
 
 -- | A data constructor. Does not include a source span and should not be
 --   transformed back.
-data ConDecl a = ConDecl (Name a) [TypeExp a]
-               | InfixConDecl (TypeExp a) (Name a) (TypeExp a)
-deriving instance EqAST a => Eq (ConDecl a)
+data ConDecl a = ConDecl (Name a) Int
```

Comment:
The `TypeExp`s are not needed since `ConDecl`s are not converted back to the original AST. The only information that we need about constructor declarations is their arity.